protocols/gossipsub/src/behaviour.rs

```diff
@@ -1623,15 +1628,22 @@ where
         }
 
         for p in px {
-            // TODO: Once signed records are spec'd: extract signed peer record if given and handle
-            // it, see https://github.com/libp2p/specs/pull/217
-            if let Some(peer_id) = p.peer_id {
+            let dial_opts = if let Some(signed_peer_record) = p.signed_peer_record {
+                // If signed envelop is invalid, return None and ignore peer exchange
```

Comment:
this only ignores peer exchange for the current peer in the list. I think we might want to halt connecting to any peer received on the exchange if this is invalid

Reply:
Then, should it collect all decoded records first, then iter and connect to all?